src/main/java/coydir/model/person/Person.java

```diff
@@ -33,13 +34,14 @@ public class Person {
      * Every field must be present and not null.
      */
     public Person(Name name, EmployeeId employeeId, Phone phone, Email email, Position position,
-            Address address, Set<Tag> tags, int numberOfLeaves) {
-        requireAllNonNull(name, employeeId, position);
+            Department department, Address address, Set<Tag> tags, int numberOfLeaves) {
+        requireAllNonNull(name, employeeId, phone, email, position, department, address);
```

Comment:
Just to clarify, only numberOfLeaves is not required to initialize an employee?

Reply:
Actually why are the rest Non Null now?

Reply:
Always has been. This is for `Person` class, so the related fields must all be present, just that the optional ones can be the `NULL` objects instead. But the objects themselves are still there.

Reply:
> Just to clarify, only numberOfLeaves is not required to initialize an employee?

Actually I suppose it should also be not null. I can go add that in.